test/Microsoft.AspNet.Cors.Test/CorsMiddlewareTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Cors.Core;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.TestHost;
+using Microsoft.Framework.DependencyInjection;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Cors.Test
+{
+    public class CorsMiddlewareTests
+    {
+        [Fact]
+        public async Task CorsRequest_MatchPolicy_SetsResponseHeaders()
+        {
+            // Arrange
+            using (var server = TestServer.Create(app =>
+            {
+                app.UseServices(services => services.AddCors());
+                app.UseCors(builder => 
+                    builder.AddOrigins("http://localhost:5001")
```

Comment:
I am just thinking should we just say `.AllowOrigins(...).AllowMethods(...)` etc?